src/platform/intel/cavs/platform.c

```diff
@@ -595,10 +596,13 @@ static void imr_layout_update(void *vector)
 	imr_layout->imr_state.header.imr_restore_vector = vector;
 	dcache_writeback_region(imr_layout, sizeof(*imr_layout));
 }
+#endif
 
 int platform_context_save(struct sof *sof)
 {
+	/* only support IMR restoring on cAVS 1.8 and onward */
+#if CAVS_VERSION >= CAVS_VERSION_1_8
 	imr_layout_update((void *)IMR_BOOT_LDR_TEXT_ENTRY_BASE);
-
+#endif
```

Comment:
The commit message makes me wonder if 
a) IMR restore feature is not supported in hardware for cAVS 1.5?
b) IMR restore feature does not currently work so it's disabled to make forward progress and deal with cAVS 1.5 later?

Which is it @keyonjie?

Reply:
> The commit message makes me wonder if
> a) IMR restore feature is not supported in hardware for cAVS 1.5?
> b) IMR restore feature does not currently work so it's disabled to make forward progress and deal with cAVS 1.5 later?
> 
> Which is it @keyonjie?

b.

Reply:
Customers are not asking for this feature on any platform. They just want reliable boot + suspend/resume.

I am not comfortable punting on cAVS 1.5, the boot flow is exactly the same for 1.5..2.5. This is taking a serious risk of missing something that will blow-up later, and having different flows to maintain. Unless IMR restore is disabled on Windows platforms we should have a plan to fix this.

Reply:
yes, I will keep following it on APL, the latest debugging makes me believe the value of that IMR region is NOT persisted, querying this to the hardware team.

Reply:
> a) IMR restore feature is not supported in hardware for cAVS 1.5?

@plbossart I think we need to say not at the moment and IIUC this was not an issue slowing APL wrt other firmwares.

Reply:
@lgirdwood let's not rush this one, I am not ready to merge this, this is one of the scariest part of the flows and we have too many unknowns left and right.

Reply:
Not going to merge until after multicore is good.

Reply:
> More information is that I found both runtime PM and S2Idle suspend works on APL, while S3 ([deep]) doesn't. 

I suspect finding this took non-negligible time so please add this as a comment in the source code not to make others pay that time (or more) again.

Reply:
Thanks @marc-hb lets comment this in the PR.